x/emissions/keeper/keeper.go

```diff
@@ -1904,111 +1888,19 @@ func (k *Keeper) PruneRecordsAfterRewards(ctx context.Context, topicId TopicId,
 }
 
 func (k *Keeper) pruneInferences(ctx context.Context, blockRange *collections.PairRange[uint64, int64]) error {
-	iter, err := k.allInferences.Iterate(ctx, blockRange)
-	if err != nil {
-		return err
-	}
-	defer iter.Close()
-
-	// Make array of keys to data to remove
-	keysToDelete := make([]collections.Pair[uint64, int64], 0)
-	for ; iter.Valid(); iter.Next() {
-		key, err := iter.KeyValue()
-		if err != nil {
-			return err
-		}
-		keysToDelete = append(keysToDelete, key.Key)
-	}
-
-	// Remove data at all keys
-	for _, key := range keysToDelete {
-		if err := k.allInferences.Remove(ctx, key); err != nil {
-			return err
-		}
-	}
-
-	return nil
+	return k.allInferences.Clear(ctx, blockRange)
```

Comment:
the type of `Inferences` is a `[]*Inference` - I hope there's no memory leaks in this, because I could see it being likely

but golang is garbage collected so maybe it's ok? i'm not sure